### PR TITLE
Dop 49 terraform GitHub org iam module

### PR DESCRIPTION
**Pull request by tudormangri10**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,33 @@
...
+|------|------|
+| [github_membership](https://registry.terraform.io/providers/integrations/github/latest/docs/resources/membership) | resource |
+| [github_team](https://registry.terraform.io/providers/integrations/github/latest/docs/resources/team) | resource |
+| [github_team_members](https://registry.terraform.io/providers/integrations/github/latest/docs/resources/team_members) | resource |
+
+# Usage Example
+
+```terraform
+module "github-iam" {
+source = "github.com/coremaker/terraform-github-org-iam.git?ref=DOP-49-terraform-github-org-iam-Module"
```
Comment (mihneastaub) [inline README.md]: A little space here

File: main.tf [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    role     = each.value
+}
+
+### Add a team to the Organization
+resource "github_team" "some_team" {
+
+  for_each = var.add_teams
+  name     = each.key
+
+  description = "Some cool team"
```
Comment (mihneastaub) [inline main.tf]: I think this should be configurable.

**Comment by mihneastaub** [comment]
Please add the followings to this repository:
- terraform pre-commit without the tflint ([docs](https://github.com/coremaker/devops-docs/blob/main/repositories/terraform_pre_commit.md))
- pull request template (https://github.com/coremaker/devops-docs/blob/main/repositories/pull_request_templates.md)

Also, it will be very useful if you will create an example directory with the usage of the module. 
Something like [this](https://github.com/terraform-aws-modules/terraform-aws-eks/tree/master/examples/complete). In this example directory, please include also the providers and everything.

File: main.tf [hunk 3]
```diff
@@ -0,0 +1,32 @@
+
```
Comment (mihneastaub) [inline main.tf]: Please format the code, you can use this command `terraform fmt` in the root dirtectory.

File: README.md (hunk 1 above)
Comment (CosminBriscaru) [reply README.md]: Maybe use source from root repo for example, without any ref/tagged version.

File: variables.tf [hunk 4]
```diff
@@ -0,0 +1,9 @@
+variable "user_list" {
```
Comment (CosminBriscaru) [inline variables.tf]: let's give this var a more meaningful name, organization_users/organization_members.

File: variables.tf [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+  default = {}
+}
+
+variable "add_teams" {
```
Comment (CosminBriscaru) [inline variables.tf]: let's give this var a more meaningful name, team_members as example

File: main.tf [hunk 6]
```diff
@@ -0,0 +1,31 @@
+### Add a user to the organization
+resource "github_membership" "membership_for_some_user" {
```
Comment (CosminBriscaru) [inline main.tf]: resource names should be more meaningful

File: main.tf [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+  for_each = var.add_teams
+  name     = each.key
+
+  description = "Some cool team"
+  privacy     = "closed"
+}
+
+### Add users from terraform.tfvars to their respective teams
+resource "github_team_members" "some_team_members" {
```
Comment (CosminBriscaru) [inline main.tf]: same here, resource names should be more meaningful.

File: main.tf [hunk 8]
```diff
@@ -0,0 +1,31 @@
+### Add a user to the organization
+resource "github_membership" "membership_for_some_user" {
+  for_each = var.user_list
+  username = each.key
+  role     = each.value
+}
+
+### Add a team to the Organization
+resource "github_team" "some_team" {
```
Comment (CosminBriscaru) [inline main.tf]: same here, resource names should be more meaningful.

File: README.md [hunk 9]
```diff
@@ -1 +1,33 @@
...
+  # Users to be added in the organization
+  user_list = {
+     "user1" = "admin"
+     "user2" = "member"
+  }
+
+  # Teams to be added in the org, along with user distribution per team
+  add_teams = {
+    test-terraform = {
+        users = ["tudormangri10", "CosminBriscaru"]
```
Comment (CosminBriscaru) [inline README.md]: Examples should not hold real users, use some placeholders with fictive accounts.

File: main.tf [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+resource "github_team_members" "some_team_members" {
+
+  for_each = var.add_teams
+  team_id  = github_team.some_team[each.key].id
+
+  dynamic "members" {
+    for_each = var.add_teams[each.key].users
+    content {
+      username = members.value
+      role     = "member"
```
Comment (CosminBriscaru) [inline main.tf:28]: How can we add a user with the role maintainer for this team?

**Review by CosminBriscaru — CHANGES_REQUESTED** [review]

File: main.tf (hunk 10 above)
Comment (tudormangri10) [reply main.tf:28]: The task description specifies users to either be members or maintainers inside the organization, not the team. Should we also add this separation inside teams?

File: examples/examples.tf [hunk 11]
```diff
@@ -0,0 +1,18 @@
+module "github-iam" {
```
Comment (mihneastaub) [inline examples/examples.tf]: Related to the examples, let's create also those files:
- `versions.tf`
- `outputs.tf`
-  `providers.tf`
- `main.tf `(renamed from `examples.tf`)

Inside of the `versions.tf` you can put something like this:
```
terraform {
  required_version = "<latest terraform version>"

  required_providers {
    github = {
      source = "integrations/github"
      version = "5.4.0"
    }
  }
}
```
Inside of the  `providers.tf` smth like this:
```
provider "github" {
  # Configuration options
}
```

The idea with the examples is that, if I want to use this module I can take the code from examples and run it.

**Review by mihneastaub — CHANGES_REQUESTED** [review]

File: examples/examples.tf (hunk 11 above)
Comment (tudormangri10) [reply examples/examples.tf]: Done

**Review by mihneastaub — APPROVED** [review]
LGTM

**Pull request merged**